src/components/common/ThreeToggle/style.module.scss

```diff
@@ -0,0 +1,110 @@
+$indicator-size: 30px;
+$switch-gap: 127%;
+$switch-gap-num: calc($switch-gap / 100%);
+$switch-start: 15%;
+$component-roundness: 34px;
+
+.switch {
+  align-items: center;
+  border-color: #9f9f9f;
+  border-radius: 34px;
+  border-style: solid;
+  border-width: 1px;
+  display: flex;
+  float: left;
+  position: relative;
+
+  label {
+    align-items: center;
+    color: #fff;
+    cursor: pointer;
+    display: flex;
+    flex-direction: row;
+    float: left;
+    height: $indicator-size * 1.2;
+    justify-content: center;
+    width: #{$indicator-size * $switch-gap-num};
+
+    input {
+      opacity: 0;
+      position: absolute;
+    }
+
+    // IF YOU WANT TO ADD BACKGROUND TO SWITCH, USE THESE
+    .switchLightMode {
+      background: #121415;
+    }
+
+    .switchDarkMode {
+      background: #fffff4;
+    }
+  }
+
+  &indicator {
+    align-self: center;
+    border-radius: 50%;
+    display: flex;
+    height: $indicator-size;
+    left: 0;
+    opacity: 50%;
+    pointer-events: none;
+    position: absolute;
+    transform: translate3d(1rem, 0, 0);
+    transition: transform 0.6s cubic-bezier(0.02, 0.94, 0.09, 0.97),
+      background 0.3s cubic-bezier(0.17, 0.67, 0.14, 1.03);
+    width: $indicator-size;
+    z-index: 1;
+  }
+
+  .indicatorLightMode {
+    background: #b8b8b8;
+  }
+
+  .indicatorDarkMode {
+    background: #9f9f9f;
+  }
```

Comment:
I see that you have a lot of functions in the component code (eg `calcIndColorStyle`) to figure out what class to apply to change the colors for light/dark mode. An alternate thing you could do, which is used in other parts of the website, is to use CSS variables:

If you look at [themes.scss](https://github.com/acmucsd/membership-portal-ui-v2/blob/main/src/styles/themes.scss), it defines CSS `var()` variables for various colors on the website, using SCSS `$` variables defined in [vars.scss](https://github.com/acmucsd/membership-portal-ui-v2/blob/main/src/styles/vars.scss), and sets them depending on the theme.

For example, for this part, you could do

```scss
// vars.scss
$light-mode-indicator: #b8b8b8;
$dark-mode-indicator: #9f9f9f;

// themes.scss
:root {
  --theme-mode-indicator: #{vars.$light-mode-indicator};
}
[data-theme='dark'] {
  --theme-mode-indicator: #{vars.$dark-mode-indicator};
}

// style.module.scss (this file)
&indicator {
  background: var(--theme-mode-indicator);
}
```

The main benefit of doing it this way is that you don't need to add any JavaScript to set the colors, so you wouldn't need the `calcIndColorStyle` function.

Reply:
Do you have a specific script I could look at for reference?

Reply:
What would I search on google to learn more about this method? Looks like it would need some adjustment to use system theme.

Reply:
Also, if we want to reuse this component, would it be better to keep all its dependencies inside its own file?

Reply:
Updated